packages/core-users/src/module/configureUsersModule.ts

```diff
@@ -374,5 +373,41 @@ export const configureUsersModule = async ({
         user: removeConfidentialServiceHashes(user),
       });
     },
+
+    addPushSubscription: async (userId, subscription, { userAgent, unsubscribeFromOtherUsers }) => {
+      if (unsubscribeFromOtherUsers) {
+        await Users.updateMany(
+          {},
+          {
+            $pull: {
+              pushSubscriptions: { keys: { p256dh: subscription?.keys?.p256dh } },
+            },
+          },
+        );
+      }
+      await Users.updateOne(
+        { _id: userId },
+        {
+          $push: {
+            pushSubscriptions: {
+              userAgent,
+              ...subscription,
+            },
+          },
+        },
+        {},
+      );
+    },
+    removePushSubscription: async (userId, p256dh) => {
+      await Users.updateOne(
+        { _id: userId },
+        {
+          $pull: {
+            pushSubscriptions: { keys: { p256dh } },
```

Comment:
I'm very sure that this code is not going to remove the push subscription, did you test this?

Reply:
Yes you're right, this works but maybe it's better to use dotted notation, else it looks like it only removes objects that look exactly like: { "keys": { "p256": "something" } } and not  { "keys": { "p256": "something" }, "userAgent": "...", "otherStuff": "..." }

Try "keys.p256dh" if it works

Reply:
@pozylon it works, I've a working rotendo version that works with this engine. shall I push that for you so you can test it,